pyccl/halos/massdef.py

```diff
@@ -84,7 +84,7 @@ class MassDef(CCLHalosObject):
             If `None`, no c(M) relation will be attached to this mass
             definition (and hence one can't translate into other definitions).
     """
-    __repr_attrs__ = ("name",)
+    __repr_attrs__ = __eq_attrs__ = ("name",)
```

Comment:
In this case `name` is built on the fly every time. Can we not build it once and store it on initialization? I think it's important if we're gonna rely so much on mdef comparisons.

Reply:
Good idea, made it a cached property, which is safe since `MassDef` objects are, and will be, immutable.